include/crm/common/xml_internal.h

```diff
@@ -400,6 +398,39 @@ void
 pcmk__xe_set_props(xmlNodePtr node, ...)
 G_GNUC_NULL_TERMINATED;
 
+/*!
+ * \internal
+ * \brief Create a new HTML element under a given parent with ID, class, and
+ *        text
+ *
+ * \param[in,out] parent   XML element that will be the new element's parent
+ *                         (\c NULL to create a new XML document with the new
+ *                         node as root)
+ * \param[in]     name     Name of new element
+ * \param[in]     id       CSS ID of new element (can be \c NULL)
+ * \param[in]     class    CSS class of new element (can be \c NULL)
+ * \param[in]     content  Text to set as the new element's content (can be
+ *                         \c NULL)
+ *
+ * \return Newly created XML element, or \c NULL on memory allocation failure
+ */
+static inline xmlNode *
+pcmk__xe_create_html(xmlNode *parent, const char *name, const char *id,
```

Comment:
Huh, I agree now that I look again. I thought we had callers outside the message output functions and adjacent code, but nope.

Reply:
I think it makes sense to take a `content` argument for this one, since every caller sets content. We can consider it a convenience function that sets content and optionally id and class. (`content` won't be required but in practice it's always passed in.)

Reply:
> Probably easiest to put it in `output_html.c`.

I think the reason it hasn't been there, is because typically every non-static function takes a `pcmk__output_t` object as the first argument. Things like this, that aren't specific to `pcmk__output_t` objects, have gone in a more generic file. Also, the HTML stuff *is* part of libxml2.

We don't have to continue with that convention, and there is one function (`pcmk__html_add_header()`) in `output_html.c` that already diverges from it.

Reply:
Yeah but I think it's more readable with the content as a separate call. There's a point at which arguments start to blur together.

Output support functions that don't use `pcmk__output_t` are fine for the `output*.c` files.

Reply:
When I push an update, the `pcmk__html_create()` calls could use a close combing through. There are a LOT of them, and all of them had to be broken into two calls with temp variables now that we no longer take a `content` argument.